### PR TITLE
feat(validations): after first hard fork

### PR DESCRIPTION
**Pull request by girazoki**

### CODE REVIEW

File: validations/src/mainnet_validations.rs [hunk 1]
```diff
@@ -30,6 +31,10 @@ pub fn in_emergency_period(
     }
 }
 
+pub fn after_first_hard_fork(epoch: Epoch) -> bool {
+    epoch > FIRST_HARD_FORK
```
Comment (tmpolaczyk) [inline validations/src/mainnet_validations.rs:35]: What's "after"?

```suggestion
    epoch >= FIRST_HARD_FORK
```

File: validations/src/mainnet_validations.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@ const FIRST_EMERGENCY_COMMITTEE: [&str; 7] = [
     "wit1etherz02v4fvqty6jhdawefd0pl33qtevy7s4z",
 ];
 
+const FIRST_HARD_FORK: u32 = "INSERT EPOCH HERE";
```
Comment (tmpolaczyk) [inline validations/src/mainnet_validations.rs:15]:
```suggestion
// FIXME(#issuenumber): decide first hard fork epoch
const FIRST_HARD_FORK: u32 = u32::MAX;
```

File: validations/src/mainnet_validations.rs (hunk 1 above)
Comment (tmpolaczyk) [reply validations/src/mainnet_validations.rs]: It was a question, I don't know if `>=` is more correct than `>`

File: validations/src/mainnet_validations.rs (hunk 2 above)
Comment (lrubiorod) [reply validations/src/mainnet_validations.rs]: Now, clippy doesn't like "absurd comparisons" :P

Comment (tmpolaczyk) [reply validations/src/mainnet_validations.rs]: Good, so we know that it's impossible for `after_first_hard_fork` to return true

Comment (tmpolaczyk) [reply validations/src/mainnet_validations.rs]: Joking aside, we can just add `#[allow(clippy::absurd_extreme_comparisons)]`

Comment (girazoki) [reply validations/src/mainnet_validations.rs]: Guys, its OK if clippy does not pass, as this branch is not gonna be merged as it is. Once we have an epoch we just type it there and clippy passes again

File: validations/src/mainnet_validations.rs (hunk 1 above)
Comment (girazoki) [reply validations/src/mainnet_validations.rs]: I am totally against triggering a semantic discussion on this haha

File: validations/src/mainnet_validations.rs (hunk 2 above)
Comment (tmpolaczyk) [reply validations/src/mainnet_validations.rs]: I thought we wanted to merge this branch and start implementing the breaking changes, and decide on the exact date when we are confident that the implementation is good

Comment (girazoki) [reply validations/src/mainnet_validations.rs]: If it is more helpful for testing purposes I can add it, never thought such a small PR was gonna generate that much discussion haaha

**Review by lrubiorod — APPROVED** [review]

File: validations/src/mainnet_validations.rs [hunk 3]
```diff
@@ -12,6 +12,8 @@ const FIRST_EMERGENCY_COMMITTEE: [&str; 7] = [
     "wit1etherz02v4fvqty6jhdawefd0pl33qtevy7s4z",
 ];
 
+// FIXME(#17): decide first hard fork epoch
```
Comment (lrubiorod) [inline validations/src/mainnet_validations.rs:15]:
```suggestion
// FIXME(#1791): decide first hard fork epoch
```

**Pull request merged**